test/test_huggingface_datasets.py

```diff
@@ -25,32 +22,24 @@
 
 
 class TestHuggingFaceHubReader(expecttest.TestCase):
-    def setUp(self):
-        self.temp_dir = create_temp_dir()
-        self.temp_files = create_temp_files(self.temp_dir)
-        self.temp_sub_dir = create_temp_dir(self.temp_dir.name)
-        self.temp_sub_files = create_temp_files(self.temp_sub_dir, 4, False)
-
-        self.temp_dir_2 = create_temp_dir()
-        self.temp_files_2 = create_temp_files(self.temp_dir_2)
-        self.temp_sub_dir_2 = create_temp_dir(self.temp_dir_2.name)
-        self.temp_sub_files_2 = create_temp_files(self.temp_sub_dir_2, 4, False)
-
-    def tearDown(self):
-        try:
-            self.temp_sub_dir.cleanup()
-            self.temp_dir.cleanup()
-            self.temp_sub_dir_2.cleanup()
-            self.temp_dir_2.cleanup()
-        except Exception as e:
-            warnings.warn(f"HuggingFace datasets was not able to cleanup temp dir due to {e}")
-
     @skipIfNoDatasets
-    def test_huggingface_hubreader(self):
-        datapipe = HuggingFaceHubReader(dataset="lhoestq/demo1", revision="main", streaming=True)
+    @patch("datasets.load_dataset")
+    def test_huggingface_hubreader(self, mock_load_dataset):
+        mock_load_dataset.return_value = datasets.Dataset.from_dict(
+            {
+                "id": ["7bd227d9-afc9-11e6-aba1-c4b301cdf627", "7bd22905-afc9-11e6-a5dc-c4b301cdf627"],
+                "package_name": ["com.mantz_it.rfanalyzer"] * 2,
+            }
+        )
+
+        datapipe = HuggingFaceHubReader("lhoestq/demo1", revision="branch", streaming=False, use_auth_token=True)
+
         elem = next(iter(datapipe))
         assert type(elem) is dict
         assert elem["package_name"] == "com.mantz_it.rfanalyzer"
+        mock_load_dataset.assert_called_with(
+            path="lhoestq/demo1", streaming=False, split="train", revision="branch", use_auth_token=True
+        )
```

Comment:
Could you please add one more line to test if there is only one element yielded from the datapipe?

Reply:
What exactly do you mean? `_get_response_from_huggingface_hub()` returns an iterator over the dataset and we look at the first element in line 37-39.

Reply:
This test validates the first output is the right one. I want to check a StopIteration should be raised when calling next over the iterator one more time